File: Find the town judge/find the town judge.java
class Solution {
    public int findJudge(int n, int[][] trust) {
        int[] outdegree = new int[n + 1];
        int[] indegree = new int[n + 1];
        Arrays.fill(outdegree, 0);
        Arrays.fill(indegree, 0);
        
        for(int i = 0; i < trust.length; i++) {
            int[] rel = trust[i];
            int from = rel[0];
            int to = rel[1];
            
            indegree[to]++;
            outdegree[from]++;
        }
        
        for(int i = 1; i <= n; i++) {
            if (indegree[i] == n - 1 && outdegree[i] == 0) {
                return i;
            }
        }
        
        return -1;
    }
}

File: Find the town judge/find the town judge.cpp
class Solution {
public:
    int findJudge(int n, vector<vector<int>>& trust) {
        vector<int> out(n + 1, 0);
        vector<int> in(n + 1, 0);
        
        for(int i = 0; i < trust.size(); i++) {
            vector<int> vec = trust[i];
            int from = vec[0];
            int to = vec[1];
            in[to]++;
            out[from]++;
        }
        
        for(int i = 1; i <= n; i++) {
            if (out[i] == 0 && in[i] == n - 1) {
                return i;
            }
        }
        
        return -1;
    }
};